bundle/permissions/utils.go

```diff
@@ -0,0 +1,81 @@
+package permissions
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/libs/diag"
+)
+
+func convert(
+	ctx context.Context,
+	bundlePermissions []resources.Permission,
+	resourcePermissions []resources.Permission,
+	resourceName string,
+	lm map[string]string,
+) []resources.Permission {
+	permissions := make([]resources.Permission, 0)
+	for _, p := range bundlePermissions {
+		level, ok := lm[p.Level]
+		// If there is no bundle permission level defined in the map, it means
+		// it's not applicable for the resource, therefore skipping
+		if !ok {
+			continue
+		}
+
+		if notifyForPermissionOverlap(ctx, p, resourcePermissions, resourceName) {
+			continue
+		}
+
+		permissions = append(permissions, resources.Permission{
+			Level:                level,
+			UserName:             p.UserName,
+			GroupName:            p.GroupName,
+			ServicePrincipalName: p.ServicePrincipalName,
+		})
+	}
+
+	return permissions
+}
+
+func isPermissionOverlap(
+	permission resources.Permission,
+	resourcePermissions []resources.Permission,
+	resourceName string,
+) (bool, diag.Diagnostics) {
+	diagnostics := make(diag.Diagnostics, 0)
```

Comment:
FWIW, this can be initialized with the zero value, like:
```
var diagnostics diag.Diagnostics
```

Extending a nil slice is fine.